### PR TITLE
Extended project reconfiguration capabilities

### LINKED ISSUE #150: Support use case: reconfiguration of Codechecker path
CodeChecker may be moved in the filesystem. The eclipse plugin should support this use case.

### PR DESCRIPTION
**Pull request by Vodorok**
Depended on #145
Depends on #159 

Project environment variables are getting reconfigured on global/project
level changing.

More to be added...
Tests also...

Resolves #150

### CODE REVIEW

**Comment by gyorb** [comment]
please rebase this PR too, thanks!

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/CommonGui.java [hunk 1]
```diff
@@ -240,7 +244,8 @@ public void run() {
             }
         });
         if (!globalGui) {
-            useGlobalSettings = config.isGlobal();
+            //TODO
```
Comment (gyorb) [inline bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/CommonGui.java]: Please add a bit more verbose comment what to do later here.

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/global/CcGlobalConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    public static CcGlobalConfiguration getInstance() {
+        if (instance == null)
+            instance = new CcGlobalConfiguration();
+        return instance;
+
+    }
+
+    /**
+     * Initializes the plug-ins internal configuration, Call this method somewhere early.
```
Comment (gyorb) [inline bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/global/CcGlobalConfiguration.java]: What does it mean early? Couldn't we call it automatically, after initialization somewhere?

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/project/CodeCheckerProject.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.osgi.service.prefs.BackingStoreException;
+
+/**
+ * This class represents the CodeChecker related side of an eclipse project.
```
Comment (gyorb) [inline bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/project/CodeCheckerProject.java]: I'm not sure I fully understand what is the `CodeChecker related side` here.

**Comment by Vodorok** [comment]
Thanks @gyorb for the review, I'll fix the issues, rebase to #159. Will do on #155 also.

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/CommonGui.java (hunk 1 above)
Comment (Vodorok) [reply bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/CommonGui.java]: Todo to remove after refactor. Thanks for the notice.

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/global/CcGlobalConfiguration.java (hunk 2 above)
Comment (Vodorok) [reply bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/global/CcGlobalConfiguration.java]: Thanks, I'll call this from the instance retrivial method. From the constructor branch, and remove from startup.

**Pull request merged**